src/main/java/com/monov/course/service/CourseService.java

```diff
@@ -15,31 +19,53 @@
     @Autowired
     private CourseRepository courseRepository;
 
-    public Course findById(Long id){
+    public CourseDTO findById(Long id){
         log.info("Inside findById method in CourseService");
-        return courseRepository.findByCourseId(id);
+        return convertToCourseDTO(courseRepository.findById(id).get());
```

Comment:
What will happen here if the findById repository method returns an empty Optional?

Reply:
Will ne handled in next PR (About exception and error handling ) which will point to this one in the description/comments